amazon/iontest/ion_test_driver.py

```diff
@@ -181,7 +183,11 @@ def execute(self, *args):
             self._executable = os.path.abspath(os.path.join(self._build_dir, self._build.execute))
         if not os.path.isfile(self._executable):
             raise ValueError('Executable for %s does not exist.' % self._name)
-        _, stderr = Popen((self._executable,) + args, stderr=PIPE, shell=COMMAND_SHELL).communicate()
+        if str.split(self.identifier, '_')[0] == 'ion-java':
+            _, stderr = Popen((("java", "-jar", self._executable,) + args),
```

Comment:
Instead of adding a special case for ion-java here, let's generalize `IonBuild` so that it supports arbitrary prefixes to the executable location for injection into the tuple provided to `Popen`. For ion-c, there is nothing because the executable executes natively, but for ion-java you'd provide, e.g. `("java", "-jar")` when creating its `IonBuild`.

Reply:
Added `prefix` field for `IonBuild`